### ISSUE TITLE
Merge CkanModule and Module to a single class

### ISSUE THREAD

**Issue by pjf** [KSP-CKAN/CKAN#1440]
Despite the fact that I was the one who wrote the original `Module` and `CkanModule` classes, I've long been puzzled as to why. After some searching I finally unearthed 9b52fa82454cad484860cfb7627d06833c6377ee, which reveals the reason they're different classes:

> We now differentiate between full modules (`CkanModules` from CKAN files) and merely bundled mods (`Modules`, which do not).

However bundled modules were *removed* from the CKAN spec well before the 1.0 release, meaning that `Module` and `CkanModule` are separate classes *for no reason at all*.

Since these cause a huge amount of bother and confusion, we should merge them into a single class. I suggest we merge them into `CkanModule`, since that's the most functional class.

This will also fix a bug I recently noticed where we'd lose install stanzas in the registry as part of serialising `InstalledModules`. While harmless to our users, it means that we lose the ability to tell exactly what metadata was used for a module's install (which is one of the reasons for stashing it).

Thank goodness for version control letting me find where the classes first split. :)

**Comment by Postremus** [comment]
IJ

----- Ursprüngliche Nachricht -----
Von: "Paul Fenwick" <notifications@github.com>
Gesendet: ‎03.‎09.‎2015 11:43
An: "KSP-CKAN/CKAN" <CKAN@noreply.github.com>
Betreff: [CKAN] Merge CkanModule and Module to a single class (#1440)

Despite the fact that I was the one who wrote the original Module and CkanModule classes, I've long been puzzled as to why. After some searching I finally unearthed 9b52fa8, which reveals the reason they're different classes:
We now differentiate between full modules (CkanModules from CKAN files) and merely bundled mods (Modules, which do not).
However bundled modules were removed from the CKAN spec well before the 1.0 release, meaning that Module and CkanModule are separate classes for no reason at all.
Since these cause a huge amount of bother and confusion, we should merge them into a single class. I suggest we merge them into CkanModule, since that's the most functional class.
This will also fix a bug I recently noticed where we'd lose install stanzas in the registry as part of serialising InstalledModules. While harmless to our users, it means that we lose the ability to tell exactly what metadata was used for a module's install (which is one of the reasons for stashing it).
Thank goodness for version control letting me find where the classes first split. :)
